spiped/main.c

```diff
@@ -275,17 +275,13 @@ main(int argc, char * argv[])
 	}
 
 	/* Resolve source address. */
-	while ((sas_s = sock_resolve(opt_s)) == NULL) {
+	while ((sa_s = sock_resolve_one(opt_s)) == NULL) {
```

Comment:
This changes the behaviour from "error out if there are no addresses" to "keep looping forever".

Reply:
Sorry, I'm not following.  I completely believe that I screwed something up, and/or we need to pass an extra argument to `sock_resolve_one` to handle this case elegantly, but I'm not seeing it.

With git master,
```
$ ./spiped/spiped -d -s 257.0.0.1:8001 -t 127.0.0.1:8000 -k /dev/null -F
spiped: Error looking up 257.0.0.1: Name does not resolve
spiped: Error resolving socket address: 257.0.0.1:8001
$ ./spiped/spiped -d -s 257.0.0.1:8001 -t 127.0.0.1:8000 -k /dev/null -F -D
spiped: Error looking up 257.0.0.1: Name does not resolve
spiped: Error looking up 257.0.0.1: Name does not resolve
spiped: Error looking up 257.0.0.1: Name does not resolve
spiped: Error looking up 257.0.0.1: Name does not resolve
^C
```

(^C indicates me manually pressing ctrl-C, of course)

With this PR,
```
$ ./spiped/spiped -d -s 257.0.0.1:8001 -t 127.0.0.1:8000 -k /dev/null -F
spiped: Error looking up 257.0.0.1: Name does not resolve
spiped: Error resolving socket address: 257.0.0.1:8001
spiped: Error resolving socket address: 257.0.0.1:8001
$ ./spiped/spiped -d -s 257.0.0.1:8001 -t 127.0.0.1:8000 -k /dev/null -F -D
spiped: Error looking up 257.0.0.1: Name does not resolve
spiped: Error resolving socket address: 257.0.0.1:8001
spiped: Error looking up 257.0.0.1: Name does not resolve
spiped: Error resolving socket address: 257.0.0.1:8001
spiped: Error looking up 257.0.0.1: Name does not resolve
spiped: Error resolving socket address: 257.0.0.1:8001
spiped: Error looking up 257.0.0.1: Name does not resolve
spiped: Error resolving socket address: 257.0.0.1:8001
^C
```

Granted, the double error message in the first case isn't ideal.  But the "loop / no loop" behaviour looks the same?

Reply:
Hmm, I was thinking of the case where we successfully get zero addresses.  (i.e. `sock_resolve` returns `sas != NULL` but `sas[0] == NULL`.)

But I'm not even sure that's possible, so let's not worry about that case.